CMakeLists.txt

```
@@ -222,6 +222,23 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/docs/getting_started.rst.in
 project(Open3D VERSION ${OPEN3D_VERSION} LANGUAGES C CXX)
 message(STATUS "Open3D ${OPEN3D_VERSION_FULL}")
 
+# FIXME: Remove this workaround once a fixed Visual Studio 16.10 version is released.
+if (BUILD_CUDA_MODULE
+    AND CMAKE_CXX_COMPILER MATCHES "MSVC"
+    AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "19.29"
+)
+    # Keep C++14 standard for unaffected C++ files, but use C++17 for CUDA files
+    set(CMAKE_CXX_STANDARD 14)
+    set(CMAKE_CUDA_STANDARD 17)
+    # Suppress warnings for deprecated C++17 functions
+    add_compile_options(
+        "$<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=/wd4996>"
+        "$<$<COMPILE_LANGUAGE:CXX>:/wd4996>">
+    )
```

`/wd4996` shouldn't be necessary for c++ code as it is still c++14.
For CUDA it is not enough and `_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING` must be defined too.
The following worked fine for me:

```suggestion
    add_compile_options($<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=/wd4996>)
    add_compile_definitions($<$<COMPILE_LANGUAGE:CUDA>:_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING>)
```